### ISSUE TITLE
Allow typevartuples in Union (Revisiting a last minute removal from pep 646)

### ISSUE THREAD

**Issue by mikeshardmind** [python/typing-council#6]
Various uses of this, including 2 which showed up in pyright's change to move support for this behind a flag for experimental features have emerged. 

- `Union[*Ts]` should be a supported construct so that typevar tuples behave identically across places where type vars are accepted.
- `Union` should not be considered deprecated, only considered as not what people should reach for first.
- This also resolves a situation where even basic things like Iterating over a generic user defined heterogenous container can't have the type of the iterator be typed properly.

Followup from, and some surrounding context [here](https://discuss.python.org/t/a-more-useful-and-less-divisive-future-for-typing/34225/99)

Implementation for this already exists in pyright, but is currently behind a flag, and is slated for potential future removal if this remains  not part of the spec.

The behavior described *prior* to the removal of this from the pep should be reincluded, and does not appear to need any major amendments in it's description, but may or may not require an additional PEP.

Edit: Annnd I'm sorry, this was supposed to be after discuss.python.org, not in parallel with it. Can close if that's the preferred way and reopen after.

**Comment by JelleZijlstra** [comment]
Thanks for bringing this up! I think it's a reasonable proposal, but it should go through discussion over on Discuss first, and there should be a concrete proposed change to the spec before the Council can make a decision. So I'm going to close this for now, and you can open a new issue here once the discussion has died down and you have a PR to amend the spec.
